### PR TITLE
Yeksi Naa Reports | Recovery Rate Report PPS level new calculations

### PR DESCRIPTION
**Pull request by dmydlo**
Hi @emord,
In this PR I have changed the way of counting Indicator 5 Report from:
Denominator
sum of data/pps_total_amt_owed
to:
Denominator
sum of:
data/pps_total_amt_owed (first visit of any month) 
AND 
data/delivery_amount_owed (of all subsequent visits of any month)

### CODE REVIEW

**Comment by emord** [comment]
looks like this causes tests to fail

**Comment by lwyszomi** [comment]
@emord tests fixed

**Comment by lwyszomi** [comment]
@emord could you kick off the build, I think to failure is not related

File: custom/intrahealth/tests/reports/test_dashboard_2.py [hunk 1]
```diff
@@ -240,16 +240,16 @@ def test_recovery_rate_by_pps_report_country_level(self):
...
                 [{'html': 'Region 1'}, {'html': 'pas de donn\xe9es'}, {'html': 'pas de donn\xe9es'},
                  {'html': 'pas de donn\xe9es'}, {'html': 'pas de donn\xe9es'}, {'html': 'pas de donn\xe9es'},
-                 {'html': '46.15%'}, {'html': '46.15%'}],
+                 {'html': '92.31%'}, {'html': '92.31%'}],
                 [{'html': 'Region Test'}, {'html': 'pas de donn\xe9es'}, {'html': 'pas de donn\xe9es'},
-                 {'html': 'pas de donn\xe9es'}, {'html': 'pas de donn\xe9es'}, {'html': '27.50%'},
-                 {'html': 'pas de donn\xe9es'}, {'html': '27.50%'}],
-                [{'html': 'Saint-Louis'}, {'html': '63.85%'}, {'html': '87.68%'}, {'html': '0.00%'},
+                 {'html': 'pas de donn\xe9es'}, {'html': 'pas de donn\xe9es'}, {'html': '191.90%'},
+                 {'html': 'pas de donn\xe9es'}, {'html': '191.90%'}],
```
Comment (emord) [inline custom/intrahealth/tests/reports/test_dashboard_2.py]: numbers over 100% seem off. Is that expected here?

File: custom/intrahealth/sqldata.py [hunk 2]
```diff
@@ -1684,7 +1684,7 @@ class RecoveryRateByPPSData(VisiteDeLOperateurDataSource):
     show_total = True
     custom_total_calculate = True
 
-    def calculate_total_row(self, data):
+    def calculate_total_row(self, data, tmp):
```
Comment (emord) [inline custom/intrahealth/sqldata.py]: It's not very clear what `tmp` means or why you need it here instead of deriving it from `data` or a database query

**Review by emord — APPROVED** [review]
kicked the build

File: custom/intrahealth/tests/reports/test_dashboard_2.py (hunk 1 above)
Comment (dmydlo) [reply custom/intrahealth/tests/reports/test_dashboard_2.py]: I have to look into that once again, as I have been making the case based on this record (Region Test) and 103% was expected (bills were slightly overpaid).

File: custom/intrahealth/sqldata.py (hunk 2 above)
Comment (dmydlo) [reply custom/intrahealth/sqldata.py]: I am going as well to clean up the naming - this one is not very useful.

**Comment by dmydlo** [comment]
Hi,
I have also stabilized tests, as the issue was with test data in which we were not able to tell which record was first as all had the same date. That lead to the problem of us being unable to tell what was the initial amount owned which is necessary for the calculation.

As well I have renamed the `tmp` variable and moved aggregating calculation to separate function.

In the end, I have allowed records only with numerator to be presented by a value rather than 'no data entered' - as I was asked in the ticket, it is done in a way that percentage value is actually an amount value. (eg. 500% would mean 500$ was paid, but nothing was due)

**Review by emord — APPROVED** [review]

**Pull request merged**